src/librustc_data_structures/stable_hasher.rs

```diff
@@ -318,6 +319,17 @@ impl<T: HashStable<CTX>, CTX> HashStable<CTX> for Vec<T> {
     }
 }
 
+impl<A, CTX> HashStable<CTX> for SmallVec<[A; 1]> where A: HashStable<CTX> {
+    #[inline]
+    fn hash_stable<W: StableHasherResult>(&self,
+                                          ctx: &mut CTX,
+                                          hasher: &mut StableHasher<W>) {
+        for item in self {
+            item.hash_stable(ctx, hasher);
+        }
```

Comment:
If you look at the `HashStable` implementation for `Vec` above, you'll see that it redirects to the slice impl because the vector length needs to be included into the hash.

The impl for `SmallVec` needs to use `(&self[..]).hash_stable(ctx, hasher);` as well.

Reply:
Done, pushed